### PR TITLE
SALTO-2162: Replace organization ids with organization names in Zendesk

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Replace organization ids with organization names in Zendesk

---

Changes: 
- Add filter which replaces organization ids with a multienv value (unique organization name)
- As most of the code is shared with user filter, I moved shared code to `replacement_utils.ts`.
- `fieldReplacer` func now supports values replacement in arrays as well. This addition also fixes a bug causing field `added_user_ids` (array field) in `user_segment` with one element to be replaced with value + arrays with multiple elements were not replaced
bug fix pics - 
image
image

Will add noise reduction for orgIds changes
Added only basic tests that i'll improve later

---
_Release Notes_: 

_Zendesk-adapter_
- References to organization ids will be replaced with organization names

---
_User Notifications_: 

_Zendesk-adapter_
- References to organization ids will be replaced with organization names

### CODE REVIEW

File: packages/zendesk-adapter/src/replacers_utils.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+const { awu } = collections.asynciterable
+
+export type ValueReplacer = (instance: InstanceElement, mapping?: Record<string, string>) => ElemID[]
+
+type FieldsParams = {
+  fieldName: string[]
+  fieldsToReplace: { name: string; valuePath?: string[] }[]
+}
+
+export const replaceConditionsAndActionsCreator = (
```
Comment (shir-reifenberg) [inline packages/zendesk-adapter/src/replacers_utils.ts:31]: I copied this as is from `user_utils.ts`

**Review by edenhassid — COMMENTED** [review]
looks good! added a few comments and questions :)

File: packages/zendesk-adapter/src/replacers_utils.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+              ? Number(newValue)
+              : newValue)
+          }
+        }
+        return [valuePath]
+      })
+  })
+)
+
+export const fieldReplacer = (fields: string[]): ValueReplacer => (instance, mapping) => (
```
[the remark this answers is not in the document]
Comment (shir-reifenberg) [reply packages/zendesk-adapter/src/replacers_utils.ts:71]: No, I changed it to support both values and arrays :)

File: packages/zendesk-adapter/src/filters/organizations.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  )).filter(isDefined)
+
+  return results.flatMap(orgResponse => orgResponse.organizations)
+}
+
+const getOrganizationsByNames = async (
+  organizationNames: string[],
+  paginator: clientUtils.Paginator,
+): Promise<Organization[]> => {
+  const paginationArgs = {
```
[the remark this answers is not in the document]
Comment (shir-reifenberg) [reply packages/zendesk-adapter/src/filters/organizations.ts:127]: Not supported :(

File: packages/zendesk-adapter/src/filters/organizations.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+type Organization = {
+  id: number
+  name: string
+}
+
+type OrganizationResponse = {
+  organizations: Organization[]
+}
+
+const ORGANIZATIONS_SCHEMA = Joi.array().items(Joi.object({
```
[the remark this answers is not in the document]
Comment (shir-reifenberg) [reply packages/zendesk-adapter/src/filters/organizations.ts:44]: Resolved IRL

**Review by edenhassid — APPROVED** [review]
🔥 🔥

**Pull request merged**